master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -243,92 +243,51 @@ public void updateRegisterWorkerMeta(
    * @throws IOException
    */
   public void writeMetaInfoToFile(File file) throws IOException, RuntimeException {
-    ObjectOutputStream out =
-        new ObjectOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
-
-    out.writeLong(estimatedPartitionSize);
-    // write registeredShuffle
-    writeSetMetaToFile(registeredShuffle, out);
-
-    // write hostnameSet
-    writeSetMetaToFile(hostnameSet, out);
-
-    // write blacklist
-    writeSetMetaToFile(blacklist, out);
-
-    // write workerLost events
-    writeSetMetaToFile(workerLostEvents, out);
-
-    // write application heartbeat time
-    out.writeInt(appHeartbeatTime.size());
-    appHeartbeatTime.forEach(
-        (app, time) -> {
-          try {
-            out.writeObject(app);
-            out.writeLong(time);
-          } catch (IOException e) {
-            throw new RuntimeException(e);
-          }
-        });
-
-    // write workerInfo
-    out.writeInt(workers.size());
-    workers.forEach(
-        workerInfo -> {
-          try {
-            out.writeObject(workerInfo);
-          } catch (IOException e) {
-            throw new RuntimeException(e);
-          }
-        });
-
-    out.writeLong(partitionTotalWritten.sum());
-    out.writeLong(partitionTotalFileCount.sum());
-    out.writeObject(appDiskUsageMetric.snapShots());
-    out.writeObject(appDiskUsageMetric.currentSnapShot());
-
+    BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(file));
```

Comment:
who is responsibility and where to close it

Reply:
> who is responsibility and where to close it

Done